File: factorial.hs
fact 0 = 1
fact n = n * fact(n-1)


fibonacci 0 = 1
fibonacci 1 = 1
fibonacci n = fibonacci(n-1) + fibonacci(n-2)

fibo prev next = [next] ++ (fibo next (prev + next))

number_case num =
    case num of
        42 -> "answer"
        5 -> "three"
        3 -> "five"
        _ -> "default"

File: factorial.py
def factorial(n):
    if n == 0:
        return 1
    else:
        return n*factorial(n-1)

def fibonacci(max_num):
    prev, current = 0,1
    while prev < max_num:
        print(prev)
        prev, current = current, prev + current

def fibo(n):
    if n == 0: return 0
    elif n == 1: return 1
    else: return fibo(n-1) + fibo(n-2)